robottelo/hosts.py

```diff
@@ -1849,8 +1849,9 @@ def get_caller():
             video_url = settings.ui.grid_url.replace(
                 ':4444', f'/videos/{ui_session.ui_session_id}.mp4'
             )
-            self.record_property('video_url', video_url)
-            self.record_property('session_id', ui_session.ui_session_id)
+            if not self.record_property and settings.ui.record_video:
```

Comment:
IMHO check for`self.record_property` to be none is not required because this could have been already populated when nested session would be runinng for a test of same satellite ?

Reply:
I wanted to prevent a scenario where the test passes, but the record property for that specific SAT instance doesn't get retrieved somehow. Omitting this condition could potentially lead to test failures. 

To address this, I can include a warning message or logging mechanism in case such a situation occurs. Another approach is to detect this issue through failing UI tests when the UI recording session cannot be obtained. This will just avoid the side effects incase.